dask/base.py

```diff
@@ -987,6 +990,37 @@ def _normalize_function(func):
             return str(func)
 
 
+def normalize_dataclass(obj):
+    # If dataclasses should not be compared, they should not be deterministically
+    # tokenized either, since the purpose of tokenization is comparision.
+    try:
+        comparable = obj.__dataclass_params__.eq
+    except AttributeError:
```

Comment:
When will we encounter this `AttributeError`? From the test coverage report, we don't appear to be hitting this code path

Reply:
We don't expect to, it's just guarding against `__dataclass_params__` not being a public API. I'll add a comment.